src/common/paddle/pricing.ts

```diff
@@ -0,0 +1,125 @@
+import { getRedisObject, setRedisObjectWithExpiry } from '../../redis';
+import { paddleInstance } from './index';
+import { ONE_HOUR_IN_SECONDS } from '../constants';
+
+export const PLUS_FEATURE_KEY = 'plus_pricing_ids';
+export const DEFAULT_PLUS_METADATA = 'plus_default';
+
+export interface PlusPricingMetadata {
+  appsId: string;
+  title: string;
+  caption?: {
+    copy: string;
+    color: string;
+  };
+  idMap: {
+    paddle: string;
+    ios: string;
+  };
+}
+
+export interface PricePreview {
+  amount: number;
+  formatted: string;
+}
+
+export interface PlusPricingPreview {
+  metadata: PlusPricingMetadata;
+  productId: string;
+  price: PricePreview & { monthly: PricePreview };
+  currency: {
+    code: string;
+    symbol: string;
+  };
+  duration: string;
+  trialPeriod?: TimePeriod | null;
+}
+
+export const getPlusPricingMetadata = async (
+  con: ConnectionManager,
+  variant: string,
+): Promise<PlusPricingMetadata[]> => {
+  const experiment = await getExperimentVariant(con, PLUS_FEATURE_KEY, variant);
+
+  if (!experiment) {
+    throw new EntityNotFoundError('ExperimentVariant not found', {
+      feature: PLUS_FEATURE_KEY,
+      variant,
+    });
+  }
+
+  try {
+    return JSON.parse(experiment.value);
+  } catch (error) {
+    throw new Error('Invalid experiment JSON value');
+  }
+};
+
+export const getPlusPricePreview = async (ctx: AuthContext, ids: string[]) => {
+  const region = ctx.region;
+  const sortedIds = ids.sort();
+
+  const hmac = createHmac('sha1', StorageTopic.Paddle);
+  hmac.update(sortedIds.toString());
+  const pricesHash = hmac.digest().toString('hex');
+
+  const redisKey = generateStorageKey(
+    StorageTopic.Paddle,
+    StorageKey.PricingPreviewPlus,
+    [pricesHash, region, ...sortedIds].join(':'),
+  );
+
+  let pricePreview: PricingPreview;
+
+  const redisResult = await getRedisObject(redisKey);
+
+  if (redisResult) {
+    pricePreview = JSON.parse(redisResult);
+  } else {
+    pricePreview = await paddleInstance?.pricingPreview.preview({
+      items: sortedIds.map((priceId) => ({
+        priceId,
+        quantity: 1,
+      })),
+      address: region ? { countryCode: region as CountryCode } : undefined,
+    });
+
+    await setRedisObjectWithExpiry(
+      redisKey,
+      JSON.stringify(pricePreview),
+      ONE_HOUR_IN_SECONDS,
+    );
+  }
+
+  return pricePreview;
+};
+
+export const getPaddleMonthlyPrice = (
+  baseAmount: number,
+  item: PricingPreviewLineItem,
+): PricePreview => {
+  const months = item.price.billingCycle?.interval === 'year' ? 12 : 1;
+  const monthlyPrice = Number(
+    (baseAmount / months).toString().match(/^-?\d+(?:\.\d{0,2})?/)?.[0],
+  );
+  const currencySymbol = item.formattedTotals.total.replace(/\d|\.|\s|,/g, '');
+  const priceFormatter = new Intl.NumberFormat('en-US', {
+    minimumFractionDigits: 2,
+  });
+
+  return {
+    amount: monthlyPrice,
+    formatted: `${currencySymbol}${priceFormatter.format(monthlyPrice)}`,
+  };
```

Comment:
These formattings are from the original pricing preview query.

Reply:
Shouldn't we also simply return the value if it's already monthly to avoid doing weird things?

Reply:
Right. Yeah, we can early return that, we can just return `null` since we have the original value from the other props already. Wdyt? @rebelchris

Reply:
Yeah sounds good :) (just safer I think)

Reply:
@capJavert pretty hard to think of cases that are not covered yet by this, but definitely open to suggestions.

@rebelchris updated the usage. We don't have to check if the cycle is `year` anymore. Using this assumes you want monthly and from the outside, I applied a ternary to check if it is monthly already, we don't need to run the function.